src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -161,11 +164,37 @@ public Query generateRollupCountsQueryForCriteriaGroupsList(
             null,
             true);
 
-    FilterVariable filterVariable =
-        new BooleanAndOrFilterVariable(
-            BooleanAndOrFilterVariable.LogicalOperator.OR,
-            criteriaGroupsList.stream().map(this::generateFilterForCriteriaGroups).toList());
+    return new Query(
+        List.of(personId), List.of(rootTable), generateFilterVariable(criteriaGroupsList));
+  }
+
+  public Query generateRowIdQueryForCohortList(List<SnapshotBuilderCohort> cohorts) {
+    List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList =
+        cohorts.stream().map(SnapshotBuilderCohort::getCriteriaGroups).toList();
+
+    FieldVariable personId =
+        new FieldVariable(
+            new FieldPointer(getRootTablePointer(), Person.PERSON_ID), rootTable, null, false);
+    FieldVariable rowId =
+        new FieldVariable(new FieldPointer(getRootTablePointer(), Person.ROW_ID), rootTable);
+
+    FilterVariable filterVariable = generateFilterVariable(criteriaGroupsList);
+
+    FilterVariable subQuery =
```

Comment:
Is a subquery needed here? E.g., instead of `SELECT row_id WHERE person_id IN (SELECT person_id WHERE <...>))` could this be `SELECT DISTINCT row_id WHERE <...>`? It seems like if we only need row IDs then we don't need to select person_ids.

If that works, then I think the count and row ID queries could be refactored so that the only difference is in their select clauses.

Reply:
yes! good idea!